api/server/router/image/image_routes.go

```diff
@@ -205,7 +205,18 @@ func (ir *imageRouter) postImagesPush(ctx context.Context, w http.ResponseWriter
 		ref = r
 	}
 
-	if err := ir.backend.PushImage(ctx, ref, metaHeaders, authConfig, output); err != nil {
+	platform, err := httputils.ParsePlatform(r.Form)
+	if err != nil {
+		return err
+	}
+
+	if platform != nil {
+		if versions.LessThan(httputils.VersionFromContext(ctx), "1.46") {
+			return errdefs.InvalidParameter(errors.New("selecting platform is not supported in API version < 1.46"))
+		}
+	}
```

Comment:
Slightly on the fence if we must do an error here or "ignore", but I think either way somewhat works. On the CLI / client side it should most likely be an error though


One thing we _should_ consider is to just check if the `platform` query-arg is set; currently we're first parsing / unmarshaling the parameter, only to produce an error; we should just error / ignore early.

Reply:
Hmmm, I think in this case it may indeed be desirable to just ignore it - mostly because the `PushOptions` already had the `Platform` field (because of it being an alias to `PullOptions`), even though it wasn't supported at all 🙈 

Changed to noop.